src/Codec/FFmpeg/Juicy.hs

```diff
@@ -110,43 +114,70 @@ instance JuicyPixelFormat PixelRGBA8 where
   juicyPixelFormat _ = avPixFmtRgba
 
 -- | Bytes-per-pixel for a JuicyPixels 'Pixel' type.
-juicyPixelStride :: forall a proxy. Pixel a => proxy a -> Int
+juicyPixelStride :: forall a proxy. (Pixel a) => proxy a -> Int
 juicyPixelStride _ =
   sizeOf (undefined :: PixelBaseComponent a) * componentCount (undefined :: a)
 
+type Metadata = NonEmpty (String, String)
+
 -- | Read frames from a video stream.
-imageReaderT :: forall m p.
-                (Functor m, MonadIO m, MonadError String m,
-                 JuicyPixelFormat p)
-             => InputSource -> m (IO (Maybe (Image p)), IO ())
-imageReaderT = fmap (first (runMaybeT . aux toJuicyImage))
-            . frameReader (juicyPixelFormat ([] :: [p]))
-  where aux g x = MaybeT x >>= MaybeT . g
+imageReaderT ::
+  forall m p.
+  ( Functor m,
+    MonadIO m,
+    MonadError String m,
+    JuicyPixelFormat p
+  ) =>
+  Bool -> 
+  InputSource ->
+  m (IO (Maybe (Image p)), IO (), VideoStreamMetadata)
+imageReaderT mdisp is = do
+  (r, c, md) <- frameReader (juicyPixelFormat ([] :: [p])) is
+  pure (aux md r, c, md)
+  where
+    aux md r = runMaybeT $ do
+      frame <- MaybeT r
+      MaybeT $ toJuicyImage mdisp (displayRotation md) frame
```

Comment:
I'm trying to understand the extra `Bool` argument we pick up in the image related functions, and I think maybe the naming is a bit confusing along this path. Above, we have `toJuicyImage rotateIfPres mdisp`, but here we have this `mdisp` parameter that is actually not the same as the `mdisp` parameter of `toJuicyImage`. I'm also a bit unclear on why we need a `Bool` in addition to the `Maybe`. Could the `Nothing` variant imply a `False` for the boolean?

Reply:
The ```Bool``` to ```imageWriterT``` allows the caller to not want the image rotated even if rotation metadata is present. I can clean up the aux function for False to be Nothing so that ```toJuicyImage``` always takes a Maybe and doesn't care about the flag.